app/controllers/incomeSources/add/IncomeSourceCheckDetailsController.scala

```diff
@@ -146,70 +148,70 @@ class IncomeSourceCheckDetailsController @Inject()(val checkDetailsView: IncomeS
 
-      val redirectUrl: (Boolean, IncomeSourceType) => String = (isAgent: Boolean, incomeSourceType: IncomeSourceType) =>
-        routes.IncomeSourceReportingMethodController.show(isAgent, incomeSourceType).url
+    val redirectUrl: (Boolean, IncomeSourceType) => String = (isAgent: Boolean, incomeSourceType: IncomeSourceType) =>
+      routes.IncomeSourceReportingMethodController.show(isAgent, incomeSourceType).url
 
-      val viewModel = getViewModel(incomeSourceType, sessionData)
+    val viewModel = getViewModel(incomeSourceType, sessionData)
 
-      viewModel match {
-        case Some(viewModel) =>
-          businessDetailsService.createRequest(viewModel) flatMap {
-            case Right(CreateIncomeSourceResponse(id)) =>
-              auditingService.extendedAudit(
-                CreateIncomeSourceAuditModel(incomeSourceType, viewModel, None, None, Some(CreateIncomeSourceResponse(id)))
-              )
-              sessionService.setMongoData(
-                sessionData.copy(
-                  addIncomeSourceData =
-                    sessionData.addIncomeSourceData.map(
-                      _.copy(
-                        incomeSourceId = Some(id)
-                      )
+    viewModel match {
+      case Some(viewModel) =>
+        businessDetailsService.createRequest(viewModel) flatMap {
+          case Right(CreateIncomeSourceResponse(id)) =>
+            auditingService.extendedAudit(
+              CreateIncomeSourceAuditModel(incomeSourceType, viewModel, None, None, Some(CreateIncomeSourceResponse(id)))
+            )
+            sessionService.setMongoData(
+              sessionData.copy(
+                addIncomeSourceData =
+                  sessionData.addIncomeSourceData.map(
+                    _.copy(
+                      incomeSourceId = Some(id)
                     )
-                )
-              ) flatMap {
-                case true => Future.successful(Redirect(redirectUrl(isAgent, incomeSourceType)))
-                case false => Future.failed(new Exception("Mongo update call was not acknowledged"))
-              }
-            case Left(ex) =>
-              auditingService.extendedAudit(
-                CreateIncomeSourceAuditModel(incomeSourceType, viewModel, Some(enums.FailureCategory.ApiFailure), Some(ex.getMessage), None)
+                  )
               )
-              Future.failed(ex)
-          }
-        case None =>
-          val agentPrefix = if (isAgent) "[Agent]" else ""
-          Logger("application").error(agentPrefix +
-            s"Unable to construct view model for $incomeSourceType")
-          Future.successful {
-            Redirect(errorRedirectUrl(isAgent, incomeSourceType))
-          }
-      }
+            ) flatMap {
+              case true => Future.successful(Redirect(redirectUrl(isAgent, incomeSourceType)))
+              case false => Future.failed(new Exception("Mongo update call was not acknowledged"))
```

Comment:
apologies if I don't fully understand the context of all the changes but wondering if we can redirect to a error page here? Instead of throwing an exception which feels unsafe.

Maybe something like this?

```
case false => Future(Redirect(errorRedirectUrl(isAgent, incomeSourceType)))
```

can refactor the lazy val into a def for the error route
```
  def errorRedirectUrl(isAgent: Boolean, incomeSourceType: IncomeSourceType): String =
    if (isAgent) routes.IncomeSourceNotAddedController.showAgent(incomeSourceType).url
    else routes.IncomeSourceNotAddedController.show(incomeSourceType).url
```